internal/bminventory/inventory.go

```diff
@@ -1729,6 +1729,16 @@ func checkPlatformWrongParamsInput(platform *models.Platform, userManagedNetwork
 		}
 	}
 
+	if cluster != nil && *cluster.Platform.Type != models.PlatformTypeBaremetal && *cluster.Platform.Type != models.PlatformTypeNone {
```

Comment:
Consider extracting this logic to a function e.g. `! (isPlatfromBM || isPlatfromNone)`